Examples/SubscriberExample/Sources/Utils/LocationAnimator/DefaultLocationAnimator.swift

```diff
@@ -0,0 +1,242 @@
+                    return
+                }
+                
+                var steps = self.createAnimationStepsFromRequest(request)
+                let expectedAnimationDuration = self.intentionalAnimationDelay + request.interval
+                            
+                // Recalculate animation duration
+                let animationStepDuration = expectedAnimationDuration / Double(steps.count)
+                
+                // Update each AnimationStep duration property
+                steps = steps.map { step -> AnimationStep in
+                    var step = step
+                    step.updateDuration(animationStepDuration)
+                    return step
+                }
+            
+                // Store last position from animation steps array
+                // In next iteration this position could be start position for the first step of the animation
+                self.previousFinalPosition = steps.last?.endPosition
+                
+                while !steps.isEmpty {
+                    self.animateStep(steps.removeFirst())
+                }
+            }
+        }.store(in: &subscriptions)
+    }
+   
+    func animateLocationUpdate(location: LocationUpdate, interval: Double) {
+        animationRequestSubject.send(AnimationRequest(locationUpdate: location, interval: interval))
+    }
+    
+    func trackablePosition(_ closure: @escaping (Position) -> Void) {
+        self.trackablePositionClosure = closure
+    }
+    
+    func fragmentaryPosition(_ closure: @escaping (Position) -> Void) {
+        self.fragmentaryPositionClosure = closure
+    }
+    
+    private func startAnimationLoop() {
+        let displayLink = CADisplayLink(target: self, selector: #selector(animationLoop))
+        displayLink.add(to: .current, forMode: .default)
+        
+        displayLinkStartTime = CFAbsoluteTimeGetCurrent()
+        self.displayLink = displayLink
+    }
+    
+    private func stopAnimationLoop() {
+        guard let displayLink = displayLink else {
+            return
+        }
+        
+        displayLink.remove(from: .current, forMode: .default)
+        displayLink.invalidate()
+    }
+    
+    private func createAnimationStepsFromRequest(_ request: AnimationRequest) -> [AnimationStep] {
+        let requestPositions = (request.locationUpdate.skippedLocations + [request.locationUpdate.location]).map { $0.toPosition() }
+        return requestPositions.enumerated().reduce([]) { [weak self] partialResult, value in
+            guard let self = self else {
+                return []
+            }
+            
+            let startPosition = value.offset == 0
+            ? self.getNewAnimationStartingPosition(locationUpdate: request.locationUpdate)
+            : requestPositions[value.offset - 1]
+            
+            return partialResult + [AnimationStep(startPosition: startPosition, endPosition: value.element, duration: self.unknownDuration)]
```

Comment:
changed here 419fae7b7f97187ca6d3d25d5329da87c25826ea

Reply:
Personally I'd still prefer not to have the `-1` being used at all, though.

Reply:
I could make it optional but I don't want to unwrap it later. What's your idea for the default value?

Reply:
My original suggestion was to have a new intermediate type that didn't include `duration`. But probably worth taking a look at [this comment thread](https://github.com/ably/ably-asset-tracking-swift/pull/289#discussion_r863069102) first, because it might be that this `AnimationStep` type goes away entirely anyway.

Reply:
@JakubJankowski Do you think this is addressable quickly? (removing the duration)

Reply:
@ikbalkaya To be honest I'm not entirely sure how introducing an intermediate type would help here. If we changed it to something like `enum Duration { case none, case some(Double) }`, we would still need to check that the value is not `none` every time we'd be trying to access it.

That's pretty much exactly what making the `duration` optional would do, but in a simpler way. We'll still need to handle unwrapping though, which Łukasz was trying to avoid.

Reply:
Maybe it's a good idea to create an issue for this instead for now, and dive back into it at a later time in a separate PR? Also if we rework the way the animations work in the future, then the problem might disappear anyway.

Reply:
Great. I think the other issue I created will naturally fix this one, So I will resolve this on behalf of @lawrence-forooghian